db/common/src/main/java/org/commonjava/indy/db/common/DefaultArtifactStoreQuery.java

```diff
@@ -550,61 +550,62 @@ public Group getGroup( final String packageType, final String name )
 
         final List<ArtifactStore> result = new ArrayList<>();
 
-        AtomicReference<IndyDataException> errorRef = new AtomicReference<>();
-        LinkedList<Group> toCheck = new LinkedList<>();
-        Set<StoreKey> seen = new HashSet<>();
-        toCheck.add( master );
+        return getMembersOrdering( master, enabled, result, includeGroups, recurseGroups );
+
+    }
 
-        while ( !toCheck.isEmpty() )
+    private List<ArtifactStore> getMembersOrdering(final Group groupRepo, final Boolean enabled, final List<ArtifactStore> result,
+                                                   final boolean includeGroups, final boolean recurseGroups) throws IndyDataException
+    {
+
+        if ( groupRepo == null || groupRepo.isDisabled() && Boolean.TRUE.equals(this.enabled) )
         {
-            Group next = toCheck.removeFirst();
+            return result;
+        }
 
-            if ( next == null || next.isDisabled() && Boolean.TRUE.equals( this.enabled ) )
-            {
-                continue;
-            }
+        Set<StoreKey> seen = new HashSet<>();
+        AtomicReference<IndyDataException> errorRef = new AtomicReference<>();
 
-            List<StoreKey> members = new ArrayList<>( next.getConstituents() );
-            if ( includeGroups )
-            {
-                result.add( next );
-            }
+        List<StoreKey> members = new ArrayList<>( groupRepo.getConstituents() );
+        if ( includeGroups )
```

Comment:
what does the includeGroups mean? it seems add the group itself to the result. I wonder what is the purpose of that? nor sure if this method only wants for concrete members?

Reply:
Yeah, it's used here: https://github.com/Commonjava/indy/blob/master/db/common/src/main/java/org/commonjava/indy/db/common/DefaultArtifactStoreQuery.java#L415

Reply:
Ok.